interfaces/builtin/fwupd.go

```diff
@@ -65,17 +65,26 @@ const fwupdPermanentSlotAppArmor = `
   # Allow fwupd to access system information
   /sys/devices/virtual/dmi/id/product_name r,
   /sys/devices/virtual/dmi/id/sys_vendor r,
+  /sys/firmware/dmi/tables/ r,
+  /sys/firmware/dmi/tables/** r,
 
   # Allow read/write access for efivarfs filesystem
   /sys/firmware/efi/efivars/ r,
   /sys/firmware/efi/efivars/** rw,
 
   # Allow write access for efi firmware updater
-  /boot/efi/EFI/ubuntu/fw/** rw,
+  /boot/efi/ r,
+  /boot/efi/** r,
+  /boot/efi/EFI/ubuntu/** rw,
 
   # Allow access from efivar library
   owner @{PROC}/@{pid}/mounts r,
+  /sys/dev/block/**/partition r,
   /sys/devices/{pci*,platform}/**/block/**/partition r,
+  /sys/devices/{pci*,platform}/**/ata_port/**/port_no r,
+  /sys/devices/**/path r,
+  /sys/devices/**/hid r,
+  /sys/devices/**/uid r,
```

Comment:
These are pretty broad. What are representative denials that prompted the rules?

Reply:
Yes, efivar is trying to parse all HW interfaces on a device via dev_probes (https://github.com/rhboot/efivar/blob/master/src/linux.c#L264) as entries of functions. For example, acpi_root_parser will indirectly call parse_acpi_hid_uid (https://github.com/rhboot/efivar/blob/master/src/linux-acpi-root.c#L122) to read these paths like https://pastebin.canonical.com/p/wPZS4xs6PP/. Unfortunately, fwupd will fail to create EFI path to report the following error If they failed:

failed to efi_generate_file_device_path(/boot/efi/EFI/ubuntu/fw/fwupd-e0f614ed-fb82-467a-a34e-71172cc07e4d.cap)

Currently, I would not prefer hacking efivar to workaround this so please advise me if we have proper expressions for rules.

Thank you.

Reply:
Probably, this is one case on a device (e.g. Dell edge 5000 I used) and we may have more combinations to face with various machines.